File: examples/v1/webhooks/go/main.go
package main

import (
	v1 "github.com/pulumi/pulumi-pulumiservice/sdk/go/pulumiservice/v1"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
	"github.com/pulumi/pulumi/sdk/v3/go/pulumi/config"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		cfg := config.New(ctx, "")
		organizationName := cfg.Get("organizationName")
		if organizationName == "" {
			organizationName = "service-provider-test-org"
		}
		secretValue := cfg.Get("secretValue")
		if secretValue == "" {
			secretValue = "shhh"
		}
		hookSuffix := cfg.Get("hookSuffix")
		if hookSuffix == "" {
			hookSuffix = "dev"
		}

		orgAll, err := v1.NewOrganizationWebhook(ctx, "orgWebhookAll", &v1.OrganizationWebhookArgs{
			OrganizationName: pulumi.String(organizationName),
			Name:             pulumi.String("org-webhook-all-" + hookSuffix),
			DisplayName:      pulumi.String("webhook-from-provider"),
			PayloadUrl:       pulumi.String("https://google.com"),
			Active:           pulumi.Bool(true),
			Secret:           pulumi.String(secretValue),
		})
		if err != nil {
			return err
		}

		orgGroups, err := v1.NewOrganizationWebhook(ctx, "orgWebhookGroups", &v1.OrganizationWebhookArgs{
			OrganizationName: pulumi.String(organizationName),
			Name:             pulumi.String("org-webhook-groups-" + hookSuffix),
			DisplayName:      pulumi.String("webhook-from-provider"),
			PayloadUrl:       pulumi.String("https://google.com"),
			Active:           pulumi.Bool(true),
			Groups:           pulumi.StringArray{pulumi.String("environments"), pulumi.String("stacks")},
			Secret:           pulumi.String(secretValue),
		})
		if err != nil {
			return err
		}

		ctx.Export("orgWebhookId", orgAll.ID())
		ctx.Export("orgWebhookGroupsId", orgGroups.ID())
		return nil
	})
}

File: examples/v1/webhooks/python/__main__.py
import pulumi
import pulumi_pulumiservice.v1 as ps_v1

config = pulumi.Config()
organization_name = config.get("organizationName") or "service-provider-test-org"
secret_value = config.get("secretValue") or "shhh"
hook_suffix = config.get("hookSuffix") or "dev"

org_webhook_all = ps_v1.OrganizationWebhook(
    "orgWebhookAll",
    organization_name=organization_name,
    name=f"org-webhook-all-{hook_suffix}",
    display_name="webhook-from-provider",
    payload_url="https://google.com",
    active=True,
    secret=secret_value,
)

org_webhook_groups = ps_v1.OrganizationWebhook(
    "orgWebhookGroups",
    organization_name=organization_name,
    name=f"org-webhook-groups-{hook_suffix}",
    display_name="webhook-from-provider",
    payload_url="https://google.com",
    active=True,
    groups=["environments", "stacks"],
    secret=secret_value,
)

pulumi.export("orgWebhookId", org_webhook_all.id)
pulumi.export("orgWebhookGroupsId", org_webhook_groups.id)
